src/api/EscargotPublic.cpp

```diff
@@ -1235,6 +1235,38 @@ void DebuggerOperationsRef::BreakpointOperations::getStackTrace(DebuggerOperatio
     }
 }
 
+void DebuggerOperationsRef::BreakpointOperations::getScopeChain(DebuggerOperationsRef::ScopeChainVector& outScopeChain)
+{
+    outScopeChain.clear();
+
+    LexicalEnvironment* lexEnv = toImpl(m_executionState)->lexicalEnvironment();
+
+    while (lexEnv) {
+        EnvironmentRecord* record = lexEnv->record();
+        ScopeType type;
+
+        if (record->isGlobalEnvironmentRecord()) {
+            type = GLOBAL_ENVIRONMENT;
+        } else if (record->isDeclarativeEnvironmentRecord()) {
+            DeclarativeEnvironmentRecord* declarativeRecord = record->asDeclarativeEnvironmentRecord();
+            if (declarativeRecord->isFunctionEnvironmentRecord()) {
+                type = FUNCTION_ENVIRONMENT;
+            } else if (record->isModuleEnvironmentRecord()) {
+                type = MODULE_ENVIRONMENT;
+            } else {
+                type = DECLARATIVE_ENVIRONMENT;
+            }
+        } else if (record->isObjectEnvironmentRecord()) {
+            type = OBJECT_ENVIRONMENT;
+        } else {
+            type = UNKNOWN_ENVIRONMENT;
```

Comment:
Don't you consider other scope type such as natvie function case?
(native function does not have `lexical environment`)

Reply:
I have forgot to add the backtrace item index to the function.

The intention of the function is returning the scope chain visible from an item of the backtrace. Is it possible to get the scope (key/value pairs) in some way for a native function?

Reply:
Native function(builtin function) has its own `ExecutionState` when it is called.
(but it does not have lexical environment)
You can check if the state is for native function or not through `m_isNativeFunctionObjectExecutionContext` as below

```c++
if (state->m_isNativeFunctionObjectExecutionContext) {
  // for native function case
}
```

Reply:
I have updated the code with the native environment. Is this code ok or do you prefer a different method?